server/mediaserver.go

```diff
@@ -789,6 +804,10 @@ func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
 					s.connectionLock.RLock()
 					if cxn, exists := s.rtmpConnections[intmid]; exists {
 						lastUsed = cxn.lastUsed
+					} else {
+						s.connectionLock.RUnlock()
+						glog.Warningf("Watchdog tried closing session for streamID=%s, which was already closed", extmid)
```

Comment:
I think also it would be good to add check for 
```go
if intmid, exists := s.internalManifests[extmid]; exists {
```
that way we will prevent two competing watchdog timers running simultaneously.